File: apps/api/routers/notificaciones.py
from fastapi import APIRouter, Depends, Query
from auth.dependencies import get_current_user
from core.database import prisma

router = APIRouter()

@router.get("")
async def list_notificaciones(
    limit: int = Query(12, le=50),
    current_user=Depends(get_current_user)
):
    try:
        items = await prisma.notificacionitem.find_many(
            where={"userId": current_user.id},
            order_by={"createdAt": "desc"},
            take=limit
        )
        unread = await prisma.notificacionitem.count(where={"userId": current_user.id, "leida": False})
        return {
            "items": [{"id": i.id, "titulo": i.titulo, "mensaje": i.mensaje, "tipo": i.tipo, "leida": i.leida, "url": i.url, "createdAt": i.createdAt.isoformat()} for i in items],
            "unread": unread
        }
    except Exception as e:
        return {"items": [], "unread": 0}

@router.get("/unread-count")
async def unread_count(current_user=Depends(get_current_user)):
    try:
        count = await prisma.notificacionitem.count(where={"userId": current_user.id, "leida": False})
        return {"count": count}
    except:
        return {"count": 0}

@router.patch("/{id}/read")
async def mark_read(id: str, current_user=Depends(get_current_user)):
    await prisma.notificacionitem.update(where={"id": id}, data={"leida": True})
    return {"ok": True}

@router.patch("/mark-all-read")
async def mark_all_read(current_user=Depends(get_current_user)):
    await prisma.notificacionitem.update_many(where={"userId": current_user.id, "leida": False}, data={"leida": True})
    return {"ok": True}

File: apps/web/lib/api/notificaciones.ts
import { api } from '../api-client';

export interface NotificacionItem {
    id: string;
    titulo: string;
    mensaje: string;
    tipo: string;
    leida: boolean;
    url?: string | null;
    createdAt: string;
}

export interface NotificacionesResponse {
    items: NotificacionItem[];
    unread: number;
}

export const notificacionesApi = {
    async listar(limit = 12): Promise<NotificacionesResponse> {
        return api.get('/notifications', { params: { limit } });
    },

    async contarNoLeidas(): Promise<{ count: number }> {
        return api.get('/notifications/unread-count');
    },

    async marcarLeida(id: string): Promise<{ ok: boolean }> {
        return api.patch(`/notifications/${id}/read`);
    },

    async marcarTodasLeidas(): Promise<{ ok: boolean }> {
        return api.patch('/notifications/mark-all-read');
    },
};
